system/Filters/Filters.php

```diff
@@ -463,20 +493,59 @@ protected function processFilters(?string $uri = null)
             // Look for inclusion rules
             if (isset($settings['before'])) {
                 $path = $settings['before'];
+
                 if ($this->pathApplies($uri, $path)) {
-                    $this->filters['before'][] = $alias;
+                    // Get arguments and clean name
+                    [$name, $arguments] = $this->getCleanName($alias);
+
+                    $this->filters['before'][] = $name;
+
+                    $this->registerArguments($name, $arguments);
                 }
             }
 
             if (isset($settings['after'])) {
                 $path = $settings['after'];
+
                 if ($this->pathApplies($uri, $path)) {
-                    $this->filters['after'][] = $alias;
+                    // Get arguments and clean name
+                    [$name, $arguments] = $this->getCleanName($alias);
+
+                    $this->filters['after'][] = $name;
+
+                    // The arguments may have already been registered in the before filter.
+                    // So disable check.
+                    $this->registerArguments($name, $arguments, false);
                 }
             }
         }
     }
 
+    /**
+     * @param string $name      filter alias
+     * @param array  $arguments filter arguments
+     * @param bool   $check     if true, check if already defined
+     */
+    private function registerArguments(string $name, array $arguments, bool $check = true): void
+    {
+        if ($arguments !== []) {
+            if ($check && array_key_exists($name, $this->arguments)) {
+                throw new ConfigException(
+                    '"' . $name . '" has already arguments: '
+                    . (($this->arguments[$name] === null) ? 'null' : implode(',', $this->arguments[$name]))
```

Comment:
How can this get a `null` argument?

Reply:
1. Set route filter with no argument.
2. Set config filter with arguments.

Added a test case: https://github.com/codeigniter4/CodeIgniter4/pull/7159/files#diff-b7622ae6db0403f01050df526b06419d30ca51b85ad2220da78fec16aaa9f3c8R285